rfcs/012-merge-rebase-strategy.md

```diff
@@ -0,0 +1,63 @@
+
+3) Rebase and merge - All the commits will be rebased and added to the base branch.
+
+### Benefits
+
+- Core contributors can choose any particular method that they see fit. Sometimes a PR would contain unnecessary 
+commits which will pollute the version history and thus a rebase or a squash makes sense. Other times it maybe that
+  all the commits in the PR adds value. 
+  
+### Downsides
+
+The main downside of this is that we are assuming that the person who made the branch/PR will keep the Git history clean
+on that branch. For this purpose we have guidelines; https://github.com/opencollective/opencollective-api/blob/main/CONTRIBUTING.md. 
+
+These guidelines propose interactive rebasing which comes with the following downsides.
+
+1) Most new contributors and especially ones who are not familiar with Git will find this confusing.
+
+2) Doing interactive re-bases for larger PRs can be time consuming when there are conflicts. Since the conflicts
+needs to be solved for each commit separately rather than as a whole. While merging seems easier since it takes the whole diff
+   and applies it in one go.
+   
+For the reasons mentioned above, this RFC proposes a solution in terms of GitHub settings in order to stick to one method when merging PRs,
+whilst giving contributors the ability to choose a method they prefer(merging, interactive rebasing etc) for their own branches/PRs.
+
+## Best Practice
+
+There's no best practice on this matter but different workflows that are suitable for different teams.
+For Open Collective projects only core team members are merging. It's hard or probably impossible to enforce all the 
+community to interactively rebase Git history. Hence, currently we are mostly relying on squashing and merging PRs which are not 
+interactively rebased and have multiple commits, or those which have merge commits from main branch. 
+
+Whether contributors use interactive re-basing, or if they choose to use simple merging, a cleaner Git history could still be 
+achieved if PRs are squashed before merging. 
+
+### Downside of this approach
+
+One downside is that when squashing and merging the person who merge needs to verify the commit title and description and make sure
+it makes sense. The GitHub interface automatically creates a commit description by concatenating all the commits in the branch. 
+Although this seems to work in most cases it's probably a good practice to see if the commit(s) make sense while merging. 
+
+## Solution
+
+[Disable `Create a merge commit` from the GitHub settings](https://docs.github.com/en/github/administering-a-repository/configuring-pull-request-merges/configuring-commit-squashing-for-pull-requests) 
+as it's probably not suitable for use. Using that option will create merge commits and will prevent a clean Git history. Additionally, suggest also disabling `Rebase and merge` as well although it could be argued to leave it in case someone would like to explicitly include
```

Comment:
I disagree to see "merge commits" as unclean. It's an established industry practice, even the default on GitHub. 

This needs to be removed or rephrased as it's an unfair argument "we want a clean git history -> we need squash and merge".

Reply:
@znarf : I agree, it really depends on preference. Done, I've removed that part and changed the wording. 😄